### PR TITLE
Fixed detection of duplicate types

### PR DESCRIPTION
**Pull request by KacerCZ**
Goes through all types and does not return on first error.
Partial fix for #5072

### CODE REVIEW

File: php/php.editor/src/org/netbeans/modules/php/editor/verification/UnusableTypesHintError.java [hunk 1]
```diff
@@ -430,18 +430,15 @@ private void checkDuplicateType(List<Expression> types) {
                 }
                 if (checkedTypes.contains(name)) {
                     createDuplicateTypeError(type, typeName);
-                    return;
```
Comment (junichi11) [inline php/php.editor/src/org/netbeans/modules/php/editor/verification/UnusableTypesHintError.java:433]: I've added it for a little performance (because the next error is shown if the error is fixed) but I don't think this fix has a big problem. So, it's OK.

File: php/php.editor/src/org/netbeans/modules/php/editor/verification/UnusableTypesHintError.java [hunk 2]
```diff
@@ -430,18 +430,15 @@ private void checkDuplicateType(List<Expression> types) {
...
                 } else if (checkedTypes.contains(Type.BOOL)) {
                     // bool|false bool|true
                     if (Type.FALSE.equals(name) || Type.TRUE.equals(name)) {
                         createDuplicateTypeError(type, typeName);
-                        return;
```
Comment (junichi11) [inline php/php.editor/src/org/netbeans/modules/php/editor/verification/UnusableTypesHintError.java:438]: Unit tests?

Comment (KacerCZ) [reply php/php.editor/src/org/netbeans/modules/php/editor/verification/UnusableTypesHintError.java:438]: I've added more test cases covering also combinations of boolean type error with duplicate class error.

File: php/php.editor/src/org/netbeans/modules/php/editor/verification/UnusableTypesHintError.java (hunk 1 above)
Comment (KacerCZ) [reply php/php.editor/src/org/netbeans/modules/php/editor/verification/UnusableTypesHintError.java:433]: Right, original code with early exit finishes faster.
If there is no error then all types will be checked anyway.

**Review by junichi11 — APPROVED** [review]
Thanks.

**Comment by KacerCZ** [comment]
Thanks for the review.
